### PR TITLE
Google Autocomplete Places API integration

### PR DESCRIPTION
**Pull request by prochetab1999**

### CODE REVIEW

File: src/components/LocationSearch.js [hunk 1]
```diff
@@ -1,12 +1,44 @@
 import { useState } from 'react'
 import FormControl from '@mui/material/FormControl'
 import TextField from '@mui/material/TextField'
+import { Autocomplete} from '@react-google-maps/api';
 
-const LocationSearch = ({type, initialLocation, onLocationChange}) => {
+const LocationSearch = ({type, initialLocation, onLocationChange,initialLocationChange,sampleText}) => {
```
Comment (hariharanweb) [inline src/components/LocationSearch.js:6]: rename initialLocationChange  as locationChange.

Comment (hariharanweb) [reply src/components/LocationSearch.js:6]: also remove sampleText, replace with what it means. Here `type`?

File: src/components/LocationSearch.js [hunk 2]
```diff
@@ -1,12 +1,44 @@
...
     const [location, setLocation] = useState(initialLocation)
+    console.log(process.env.REACT_APP_API_KEY);
+  const [autocomplete,setAutoComplete]=useState(null);
+  
+  const onLoad =(autocomplete)=> {
+    console.log('autocomplete: ', autocomplete)
```
Comment (hariharanweb) [inline src/components/LocationSearch.js:12]: Remove console.logs

File: src/components/LocationSearch.js [hunk 3]
```diff
@@ -1,12 +1,44 @@
...
+      let locationObj={};
+      locationObj.display=place?.name;
+      locationObj.latLong=place?.geometry?.location.lat()+","+place?.geometry?.location.lng();
+      console.log(locationObj);
+      setLocation(locationObj);
+      initialLocationChange(locationObj);
+    }
+  }
     return (
+
```
Comment (hariharanweb) [inline src/components/LocationSearch.js:32]: Did it work? You cant have 2 root elements in render

File: src/screens/SearchScreen.js [hunk 4]
```diff
@@ -37,14 +31,16 @@ const SearchScreen = () => {
         }
         const response = await Api.post('/search', data)
         navigate('/search', {state: {...response}});
+        // console.log(fromLocation);
```
Comment (hariharanweb) [inline src/screens/SearchScreen.js:34]: Remove

File: src/screens/SearchScreen.js [hunk 5]
```diff
@@ -37,14 +31,16 @@ const SearchScreen = () => {
...
+        // console.log(toLocation);
     }
     return (
         <Grid container paddingX={4}>
             <Typography variant="h4" gutterBottom paddingY={1}>
                 ONDC Sample App
             </Typography>
-            <LocationSearch type="From" initialLocation={fromLocation}/>
-            <LocationSearch type="To" initialLocation={toLocation}/>
+            <LocationSearch type="From" initialLocation={fromLocation} initialLocationChange={setFromLocation} sampleText="Enter Pickup Point"/>
```
Comment (hariharanweb) [inline src/screens/SearchScreen.js:42]: Rename as placeholder

Comment (prochetab1999) [reply src/screens/SearchScreen.js]: Updated

File: src/screens/SearchScreen.js (hunk 4 above)
Comment (prochetab1999) [reply src/screens/SearchScreen.js]: Updated

File: src/components/LocationSearch.js (hunk 2 above)
Comment (prochetab1999) [reply src/components/LocationSearch.js]: Updated

File: src/components/LocationSearch.js (hunk 1 above)
Comment (prochetab1999) [reply src/components/LocationSearch.js]: updated

File: src/components/LocationSearch.js (hunk 3 above)
Comment (japneetbhatia) [reply src/components/LocationSearch.js]: Screenshot 2022-12-19 at 4 01 44 PM

There is only 1 root element FormControl

**Pull request merged**